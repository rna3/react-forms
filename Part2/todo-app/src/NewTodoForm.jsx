import React, { useState } from 'react';

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    if (task.trim()) {
      addTodo(task);
      setTask('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="New task"
        value={task}
        onChange={(e) => setTask(e.target.value)}
      />
      <button>Add</button>
    </form>
  );
}

export default NewTodoForm;
