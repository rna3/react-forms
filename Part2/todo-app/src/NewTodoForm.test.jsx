import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

test("renders NewTodoForm without crashing", () => {
  render(<NewTodoForm addTodo={() => {}} />);
});

test("matches snapshot for NewTodoForm", () => {
    const { asFragment } = render(<NewTodoForm addTodo={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });