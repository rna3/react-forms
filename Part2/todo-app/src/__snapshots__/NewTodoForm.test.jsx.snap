// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for NewTodoForm 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="New task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
